src/api/spec/features/webui/announcements_spec.rb

```diff
@@ -0,0 +1,35 @@
+require 'browser_helper'
+
+RSpec.feature 'Announcements for users', type: :feature, js: true do
```

Comment:
A feature test is rather expensive for this. I think a controller test testing that `@pending_announcement` is assigned if it's pending and not if not should be enough. The rest is not really test worthy IMHO...

Reply:
But it also checks that, after clicking on "Got it",  the button disappears. Also checks that ToS is shown for `nobody' but not the button.

Reply:
And of course that `Webui::Users::AnnouncementsController#create` works. But as this is only standard rails stuff I wouldn't even bother with this.

Reply:
How about having one feature test that ensures that users see the ToS dialog and are able to confirm it? It would be odd if this stops working and we don't notice

Reply:
Okay with me.